python/pyspark/sql/types.py

```diff
@@ -107,7 +107,8 @@ class NullType(DataType, metaclass=DataTypeSingleton):
 
     The data type representing None, used for the types that cannot be inferred.
     """
-    pass
+    def simpleString(self):
```

Comment:
Can you override `typeName` instead, and see if it work? Also it would be great if we can remove this line: https://github.com/apache/spark/blob/master/python/pyspark/sql/tests/test_types.py#L514
to test round trip in PySpark type parsing.

Reply:
(Python version of `typeName` is slightly different with Scala side)

Reply:
sure thing. I will double confirm changing `typeName` in python side will not affect the `Datatype.json` because that may lead to compatibility concerns.

Reply:
the json format there is internal purpose. It's not supposed to be stored outside so it should be fine. I am asking this because usage with `typeName` is documented, for exmaple, here:

https://github.com/apache/spark/blob/2537fe8cbaf49070137d4b5bc39af078b306c4c8/python/pyspark/sql/session.py#L51-L52

If we're worried about anything related to it, we could rather update the docs instead.